hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetTable.java

```diff
@@ -873,45 +880,57 @@ private void assignDsToTarget(final DragAndDropEvent event) {
             return;
         }
 
-        final TargetIdName createTargetIdName = new TargetIdName(target.get());
-        final List<DistributionSet> findDistributionSetById = distributionSetManagement
-                .get(new HashSet<>(Arrays.asList(idToSelect)));
-
-        if (findDistributionSetById.isEmpty()) {
+        final List<DistributionSet> distributionSets = distributionSetManagement.get(dsIds);
+        if (distributionSets.isEmpty()) {
             getNotification().displayWarning(getI18n().getMessage(DISTRIBUTIONSET_NOT_EXISTS));
             return;
         }
-        final DistributionSet distributionSetToAssign = findDistributionSetById.get(0);
-        addNewTargetToAssignmentList(createTargetIdName, distributionSetToAssign);
-        openConfirmationWindowForAssignment(distributionSetToAssign.getName(), createTargetIdName.getTargetName());
-    }
-
-    private void openConfirmationWindowForAssignment(final String distributionNameToAssign, final String targetName) {
-        confirmDialog = new ConfirmationDialog(getI18n().getMessage(CAPTION_ENTITY_ASSIGN_ACTION_CONFIRMBOX),
-                getI18n().getMessage(MESSAGE_CONFIRM_ASSIGN_ENTITY, distributionNameToAssign, "target", targetName),
-                getI18n().getMessage(UIMessageIdProvider.BUTTON_OK),
-                getI18n().getMessage(UIMessageIdProvider.BUTTON_CANCEL), ok -> {
-                    if (ok && isMaintenanceWindowValid(maintenanceWindowLayout, getNotification())) {
-                        saveAllAssignments(managementUIState, actionTypeOptionGroupLayout, maintenanceWindowLayout,
-                                deploymentManagement, getNotification(), getEventBus(), getI18n(), this);
-                    } else {
-                        managementUIState.getAssignedList().clear();
-                    }
-                }, createAssignmentTab(actionTypeOptionGroupLayout, maintenanceWindowLayout, saveButtonToggle(),
-                        getI18n(), uiProperties),
-                UIComponentIdProvider.DIST_SET_TO_TARGET_ASSIGNMENT_CONFIRM_ID);
+        openConfirmationWindowForAssignments(target.get(), distributionSets);
+    }
+
+    private Set<Long> filterDistributionSetsToAssign(final Set<Long> ids) {
+        if (isMultiAssignmentsEnabled()) {
```

Comment:
Better:

```java
if (ids.isEmpty()) {
    return Collections.emptySet();
}
if (isMultiAssignmentsEnabled()) {
    return new HashSet<>(ids);
}
return Collections.singleton(ids.iterator().next());
```
Otherwise you'd create a new `HashSet` even `ids` is empty in multi-assignment mode